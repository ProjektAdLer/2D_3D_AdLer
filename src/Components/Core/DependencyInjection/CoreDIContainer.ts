import { LoadWorldController } from "../Presentation/React/LoadWorldButton/LoadWorldController";
import ILoadWorldController from "../Presentation/React/LoadWorldButton/ILoadWorldController";
import { Container } from "inversify";
import CORE_TYPES from "./CoreTypes";
import ISceneView from "../Presentation/Babylon/SceneManagement/ISceneView";
import SceneView from "../Presentation/Babylon/SceneManagement/SceneView";
import SceneViewModel from "../Presentation/Babylon/SceneManagement/SceneViewModel";
import ScenePresenter from "../Presentation/Babylon/SceneManagement/ScenePresenter";
import MainScene from "../Presentation/Babylon/SceneManagement/MainScene";
import ICreateSceneClass from "../Presentation/Babylon/SceneManagement/ICreateSceneClass";
import IScenePresenter from "../Presentation/Babylon/SceneManagement/IScenePresenter";
import infrastructureDIContainer from "./InfrastructureDIContainer";
import useCaseDIContainer from "./UseCases/UseCaseDIConatiner";
import PortsDIContainer from "./Ports/PortsDIContainer";
import BuilderDIContainer from "./Builders/BuilderDIContainer";
import INavigation from "../Presentation/Babylon/Navigation/INavigation";
import Navigation from "../Presentation/Babylon/Navigation/Navigation";
import NavigationConfiguration from "../Presentation/Babylon/Navigation/NavigationConfiguration";

var CoreDIContainer = new Container();

// Scene
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind(SceneViewModel).toSelf().inSingletonScope();
CoreDIContainer.bind<IScenePresenter>(CORE_TYPES.IScenePresenter)
  .to(ScenePresenter)
  .inSingletonScope();

// Navigation
CoreDIContainer.bind<INavigation>(CORE_TYPES.INavigation)
  .to(Navigation)
  .inSingletonScope();
CoreDIContainer.bind(NavigationConfiguration).toSelf();

// Controllers
CoreDIContainer.bind<ILoadWorldController>(CORE_TYPES.ILoadWorldController)
  .to(LoadWorldController)
  .inSingletonScope();

// bind other CreateSceneClass here for testing puposes -MK
CoreDIContainer.bind<ICreateSceneClass>(CORE_TYPES.ICreateSceneClass).to(
  MainScene
);

CoreDIContainer.load(infrastructureDIContainer);
CoreDIContainer.load(useCaseDIContainer);
CoreDIContainer.load(PortsDIContainer);
CoreDIContainer.load(BuilderDIContainer);

export default CoreDIContainer;
