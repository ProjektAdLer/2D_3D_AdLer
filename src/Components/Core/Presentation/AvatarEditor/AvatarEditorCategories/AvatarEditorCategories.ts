export const OAvatarEditorCategory = {
  HAIR: 0,
  FACE: 1,
  ACCESSOIRE: 2,
  CLOTHING: 3,
  BODY: 4,
} as const;

export type AvatarEditorCategory =
  (typeof OAvatarEditorCategory)[keyof typeof OAvatarEditorCategory];
