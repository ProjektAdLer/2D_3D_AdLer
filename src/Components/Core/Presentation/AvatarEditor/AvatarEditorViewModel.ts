import Observable from "src/Lib/Observable";
import { AvatarColor } from "../../Domain/AvatarModels/AvatarColorPalette";
import {
  AvatarHairModels,
  AvatarBeardModels,
  AvatarHeadgearModels,
  AvatarGlassesModels,
  AvatarShirtModels,
  AvatarPantsModels,
  AvatarShoesModels,
  AvatarBackpackModels,
  AvatarOtherModels,
} from "../../Domain/AvatarModels/AvatarModelTypes";

export default class AvatarEditorViewModel {
  // Face
  eyebrows: Observable<number> = new Observable<number>(0);
  eyes: Observable<number> = new Observable<number>(0);
  nose: Observable<number> = new Observable<number>(0);
  mouth: Observable<number> = new Observable<number>(0);

  // Hair
  hair: Observable<AvatarHairModels> = new Observable<AvatarHairModels>("none");
  beard: Observable<AvatarBeardModels> = new Observable<AvatarBeardModels>(
    "none",
  );
  hairColor: Observable<AvatarColor> = new Observable<AvatarColor>({
    id: 33,
    nameKey: "Brown 2",
    hexColor: "#4b2a1a",
  });

  // Accessories
  headgear: Observable<AvatarHeadgearModels> =
    new Observable<AvatarHeadgearModels>("none");
  glasses: Observable<AvatarGlassesModels> =
    new Observable<AvatarGlassesModels>("none");
  backpack: Observable<AvatarBackpackModels> =
    new Observable<AvatarBackpackModels>("none");
  other: Observable<AvatarOtherModels> = new Observable<AvatarOtherModels>(
    "none",
  );

  // Clothes
  shirt: Observable<AvatarShirtModels>;
  shirtColor: Observable<AvatarColor>;
  pants: Observable<AvatarPantsModels>;
  pantsColor: Observable<AvatarColor>;
  shoes: Observable<AvatarShoesModels>;
  shoesColor: Observable<AvatarColor>;

  // Body
  skinColor: Observable<AvatarColor> = new Observable<AvatarColor>({
    id: 33,
    nameKey: "Brown 2",
    hexColor: "#4b2a1a",
  });
  roundness: Observable<number>; // 0-1 morph target weight

  // --------------------------------------------------------------
  // UI Component Visibility
  // --------------------------------------------------------------

  uiVisiblity: AvatarEditorUI = {
    hairMenu: {
      hairstyles: new Observable<boolean>(true),
      beards: new Observable<boolean>(true),
    },
    faceMenu: {
      eyebrows: new Observable<boolean>(true),
      eyes: new Observable<boolean>(false),
      noses: new Observable<boolean>(false),
      mouths: new Observable<boolean>(false),
    },
    accessoireMenu: {
      headGear: new Observable<boolean>(true),
      glasses: new Observable<boolean>(false),
      backpack: new Observable<boolean>(false),
      other: new Observable<boolean>(false),
    },
    clothingMenu: {
      shirts: new Observable<boolean>(true),
      pants: new Observable<boolean>(false),
      shoes: new Observable<boolean>(false),
    },
  };
}

export interface AvatarConfigHairIconVisibility {
  hairstyles: Observable<boolean>;
  beards: Observable<boolean>;
}

export interface AvatarConfigFaceIconVisibility {
  eyebrows: Observable<boolean>;
  eyes: Observable<boolean>;
  noses: Observable<boolean>;
  mouths: Observable<boolean>;
}

export interface AvatarConfigAccessoireIconVisibility {
  headGear: Observable<boolean>;
  glasses: Observable<boolean>;
  backpack: Observable<boolean>;
  other: Observable<boolean>;
}

export interface AvatarConfigClothingIconVisibility {
  shirts: Observable<boolean>;
  pants: Observable<boolean>;
  shoes: Observable<boolean>;
}

export interface AvatarEditorUI {
  hairMenu: AvatarConfigHairIconVisibility;
  faceMenu: AvatarConfigFaceIconVisibility;
  accessoireMenu: AvatarConfigAccessoireIconVisibility;
  clothingMenu: AvatarConfigClothingIconVisibility;
}
