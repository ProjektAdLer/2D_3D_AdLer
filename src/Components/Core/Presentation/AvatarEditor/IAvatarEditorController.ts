import AvatarConfigTO from "../../Application/DataTransferObjects/AvatarConfigTO";

export default interface IAvatarEditorController {
  onAvatarConfigChanged(changes: Partial<AvatarConfigTO>): void;
  saveAvatarConfig(): void;
  resetAvatarConfig(): void;
}
