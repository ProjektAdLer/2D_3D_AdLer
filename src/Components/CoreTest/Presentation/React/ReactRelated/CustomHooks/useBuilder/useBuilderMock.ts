import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
