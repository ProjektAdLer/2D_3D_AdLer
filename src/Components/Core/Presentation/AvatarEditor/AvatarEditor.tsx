import IAvatarEditorController from "./IAvatarEditorController";
import AvatarEditorViewModel from "./AvatarEditorViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import React, { useState } from "react";
import MenuHeaderBar from "~ReactComponents/GeneralComponents/MenuHeaderBar/MenuHeaderBar";
import AvatarEditorCategoryTabButton from "./AvatarEditorCategories/AvatarEditorCategoryTabButton";
import {
  AvatarEditorCategory,
  OAvatarEditorCategory,
} from "./AvatarEditorCategories/AvatarEditorCategories";
import AvatarEditorHairCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorHairCategory";
import AvatarEditorFaceCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorFaceCategory";
import AvatarEditorPreview from "./AvatarEditorPreview/AvatarEditorPreview";
import LoadingScreen from "~ReactComponents/GeneralComponents/LoadingScreen/LoadingScreen";

export default function AvatarEditor() {
  const [viewModel, controller] = useBuilder<
    AvatarEditorViewModel,
    IAvatarEditorController
  >(BUILDER_TYPES.IAvatarEditorBuilder);

  const [activeTab, setActiveTab] = useState<AvatarEditorCategory>(
    OAvatarEditorCategory.HAIR,
  );

  if (!viewModel || !controller) return null;

  return (
    <React.Fragment>
      <div className="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden">
        <MenuHeaderBar
          location="editor"
          className="self-center w-full p-2 font-semibold"
        />
        <div className="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 grow lg:rounded-lg">
          {/* Categories (Left Side) */}

          <div className="flex flex-col portrait:order-2">
            {/* Category Tabs */}
            {/* Category Header */}
            <div className="flex flex-row items-center justify-center">
              <div className="flex flex-col items-center justify-center">
                <div className="text-2xl font-bold text-darkblue">
                  {activeTab === OAvatarEditorCategory.HAIR && "Haare"}
                  {activeTab === OAvatarEditorCategory.FACE && "Gesicht"}
                  {activeTab === OAvatarEditorCategory.ACCESSOIRE &&
                    "Accessoires"}
                  {activeTab === OAvatarEditorCategory.CLOTHING && "Kleidung"}
                </div>
              </div>
            </div>
            <div className="flex flex-row justify-center p-2 space-x-4 ">
              {Object.values(OAvatarEditorCategory)
                .filter((category) => typeof category === "number")
                .map((category) => (
                  <AvatarEditorCategoryTabButton
                    key={category}
                    category={category as AvatarEditorCategory}
                    active={activeTab === category}
                    onClick={() => setActiveTab(category)}
                  />
                ))}
            </div>

            {/* Category Contents */}
            <div className="p-4 max-h-[77vh] portrait:max-h-[30vh] overflow-auto">
              {activeTab === OAvatarEditorCategory.HAIR && (
                <AvatarEditorHairCategory controller={controller} />
              )}
              {activeTab === OAvatarEditorCategory.FACE && (
                <AvatarEditorFaceCategory controller={controller} />
              )}
            </div>
          </div>

          {/* Avatar Preview (Right Side) */}
          <div className="p-4">
            <AvatarEditorPreview className="relative w-full h-full" />
          </div>
        </div>
        <LoadingScreen />
      </div>
    </React.Fragment>
  );
}
