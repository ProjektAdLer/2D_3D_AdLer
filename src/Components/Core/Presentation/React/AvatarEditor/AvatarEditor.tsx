import IAvatarEditorController from "./IAvatarEditorController";
import AvatarEditorViewModel from "./AvatarEditorViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import React, { useState } from "react";
import MenuHeaderBar from "~ReactComponents/GeneralComponents/MenuHeaderBar/MenuHeaderBar";
import AvatarEditorCategoryTabButton from "./AvatarEditorCategories/AvatarEditorCategoryTabButton";
import {
  AvatarEditorCategory,
  OAvatarEditorCategory,
} from "./AvatarEditorCategories/AvatarEditorCategories";
import AvatarEditorHairCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorHairCategory";
import AvatarEditorFaceCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorFaceCategory";

const AvatarEditorCategoryContents = {
  [OAvatarEditorCategory.HAIR]: AvatarEditorHairCategory,
  [OAvatarEditorCategory.FACE]: AvatarEditorFaceCategory,
};

export default function AvatarEditor() {
  const [viewModel, controller] = useBuilder<
    AvatarEditorViewModel,
    IAvatarEditorController
  >(BUILDER_TYPES.IAvatarEditorBuilder);

  const [activeTab, setActiveTab] = useState<AvatarEditorCategory>(
    OAvatarEditorCategory.HAIR,
  );

  if (!viewModel || !controller) return null;

  return (
    <React.Fragment>
      <div className="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden">
        <MenuHeaderBar
          location="editor"
          className="self-center w-full p-2 font-semibold"
        />
        <div className="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg">
          {/* Categories (Left Side) */}
          <div className="portrait:order-2">
            {/* Category Tabs */}
            <div className="flex flex-row justify-center space-x-4 p-4">
              {Object.values(OAvatarEditorCategory)
                .filter((category) => typeof category === "number")
                .map((category) => (
                  <AvatarEditorCategoryTabButton
                    className=""
                    key={category}
                    category={category as AvatarEditorCategory}
                    active={activeTab === category}
                    onClick={() => setActiveTab(category)}
                  />
                ))}
            </div>

            {/* Category Contents */}
            <div className="p-4">
              {AvatarEditorCategoryContents[activeTab]({
                controller,
              })}
            </div>
          </div>

          {/* Avatar Preview (Right Side) */}
          <div className="p-4">AVATAR THUMBNAIL</div>
        </div>
      </div>
    </React.Fragment>
  );
}
