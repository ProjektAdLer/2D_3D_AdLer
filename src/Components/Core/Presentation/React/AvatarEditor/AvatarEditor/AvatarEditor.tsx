import IAvatarEditorController from "./IAvatarEditorController";
import AvatarEditorViewModel from "./AvatarEditorViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import React from "react";
import MenuHeaderBar from "~ReactComponents/GeneralComponents/MenuHeaderBar/MenuHeaderBar";

export default function AvatarEditor() {
  const [viewModel, controller] = useBuilder<
    AvatarEditorViewModel,
    IAvatarEditorController
  >(BUILDER_TYPES.IAvatarEditorBuilder);

  if (!viewModel || !controller) return null;

  return (
    <React.Fragment>
      <div className="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden">
        <div className="grid order-2 grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg">
          <div className="col-start-1">links</div>
          <div className="col-start-2">rechts</div>
        </div>
        <MenuHeaderBar
          location="editor"
          className="self-center order-1 w-full p-2 font-semibold"
        />
      </div>
    </React.Fragment>
  );
}
