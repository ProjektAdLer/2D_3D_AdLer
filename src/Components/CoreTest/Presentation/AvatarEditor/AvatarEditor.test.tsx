import React from "react";
import { act, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import AvatarEditor from "../../../Core/Presentation/AvatarEditor/AvatarEditor";
import { Provider } from "inversify-react";
import CoreDIContainer from "../../../Core/DependencyInjection/CoreDIContainer";
import { OAvatarEditorCategory } from "../../../Core/Presentation/AvatarEditor/AvatarEditorCategories/AvatarEditorCategories";
import useBuilderMock from "../React/ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import AvatarEditorViewModel from "../../../Core/Presentation/AvatarEditor/AvatarEditorViewModel";
import IAvatarEditorController from "../../../Core/Presentation/AvatarEditor/IAvatarEditorController";
import mock from "jest-mock-extended/lib/Mock";

jest.mock(
  "../../../Core/Presentation/AvatarEditor/AvatarEditorPreview/AvatarEditorPreview",
  () => "MockedAvatarEditorPreview",
);

describe("AvatarEditor", () => {
  test("should renders", () => {
    useBuilderMock([
      new AvatarEditorViewModel(),
      mock<IAvatarEditorController>(),
    ]);

    const { container } = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditor />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  test("doesn't render without viewModel or controller", () => {
    useBuilderMock([undefined, undefined]);

    const { container } = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditor />
      </Provider>,
    );

    expect(container).toBeEmptyDOMElement();
  });

  test.each([[OAvatarEditorCategory.HAIR], [OAvatarEditorCategory.FACE]])(
    "should render %s category when tab button is clicked",
    (category) => {
      useBuilderMock([
        new AvatarEditorViewModel(),
        mock<IAvatarEditorController>(),
      ]);

      const result = render(
        <Provider container={CoreDIContainer}>
          <AvatarEditor />
        </Provider>,
      );

      const tabButton = result.getByTestId(
        `avatar-editor-category-tab-${category}`,
      );

      act(() => {
        fireEvent.click(tabButton);
      });

      expect(result.container).toMatchSnapshot();
    },
  );
});
