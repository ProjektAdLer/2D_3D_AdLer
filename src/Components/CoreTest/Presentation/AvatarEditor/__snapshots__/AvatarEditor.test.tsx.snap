// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarEditor should render 0 category when tab button is clicked 1`] = `
<div>
  <div
    class="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden"
  >
    <div
      class="self-center w-full p-2 font-semibold flex place-content-stretch"
    >
      <div
        class="flex items-center w-1/2 place-content-stretch justify-self-start "
      >
        <div
          class=""
        >
          <button
            class="mr-4 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
          >
            <img
              alt="Home Icon"
              class="w-10 xl:w-12 "
              src="home.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="flex justify-center w-full"
      >
        <div
          class="text-xl truncate lg:text-4xl font-[roboto] flex gap-2 m-2 w-fit h-fit text-adlerdarkblue"
        >
          Avatar-Editor
        </div>
      </div>
      <div
        class="flex justify-end w-1/2 "
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
            >
              <img
                alt="Help Icon"
                class="w-10 xl:w-12"
                src="help.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2  grow lg:rounded-lg"
    >
      <div
        class="flex flex-col portrait:order-2 "
      >
        <div
          class="flex flex-row justify-center space-x-4 p-4 "
        >
          <div
            class=""
          >
            <button
              class="!bg-adleryellow translate-x-[1px] translate-y-[1px] !border-b-2 !border-r-2 !border-transparent justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="avatar-editor-category-tab-0"
            >
              <img
                alt=""
                class="h-8 lg:h-12"
                src="hair.svg"
              />
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="avatar-editor-category-tab-1"
            >
              <img
                alt=""
                class="h-8 lg:h-12"
                src="face.svg"
              />
            </button>
          </div>
        </div>
        <div
          class="p-4 max-h-[77vh] portrait:max-h-[30vh] overflow-auto"
        >
          <mockedavatareditorhaircategory
            controller="undefined"
          />
        </div>
      </div>
      <div
        class="p-4"
      >
        <mockedavatareditorpreview
          class="relative w-full h-full"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarEditor should render 1 category when tab button is clicked 1`] = `
<div>
  <div
    class="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden"
  >
    <div
      class="self-center w-full p-2 font-semibold flex place-content-stretch"
    >
      <div
        class="flex items-center w-1/2 place-content-stretch justify-self-start "
      >
        <div
          class=""
        >
          <button
            class="mr-4 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
          >
            <img
              alt="Home Icon"
              class="w-10 xl:w-12 "
              src="home.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="flex justify-center w-full"
      >
        <div
          class="text-xl truncate lg:text-4xl font-[roboto] flex gap-2 m-2 w-fit h-fit text-adlerdarkblue"
        >
          Avatar-Editor
        </div>
      </div>
      <div
        class="flex justify-end w-1/2 "
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
            >
              <img
                alt="Help Icon"
                class="w-10 xl:w-12"
                src="help.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2  grow lg:rounded-lg"
    >
      <div
        class="flex flex-col portrait:order-2 "
      >
        <div
          class="flex flex-row justify-center space-x-4 p-4 "
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="avatar-editor-category-tab-0"
            >
              <img
                alt=""
                class="h-8 lg:h-12"
                src="hair.svg"
              />
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="!bg-adleryellow translate-x-[1px] translate-y-[1px] !border-b-2 !border-r-2 !border-transparent justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="avatar-editor-category-tab-1"
            >
              <img
                alt=""
                class="h-8 lg:h-12"
                src="face.svg"
              />
            </button>
          </div>
        </div>
        <div
          class="p-4 max-h-[77vh] portrait:max-h-[30vh] overflow-auto"
        >
          <mockedavatareditorfacecategory
            controller="undefined"
          />
        </div>
      </div>
      <div
        class="p-4"
      >
        <mockedavatareditorpreview
          class="relative w-full h-full"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarEditor should renders 1`] = `
<div>
  <div
    class="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden"
  >
    <div
      class="self-center w-full p-2 font-semibold flex place-content-stretch"
    >
      <div
        class="flex items-center w-1/2 place-content-stretch justify-self-start "
      >
        <div
          class=""
        >
          <button
            class="mr-4 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
          >
            <img
              alt="Home Icon"
              class="w-10 xl:w-12 "
              src="home.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="flex justify-center w-full"
      >
        <div
          class="text-xl truncate lg:text-4xl font-[roboto] flex gap-2 m-2 w-fit h-fit text-adlerdarkblue"
        >
          Avatar-Editor
        </div>
      </div>
      <div
        class="flex justify-end w-1/2 "
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
            >
              <img
                alt="Help Icon"
                class="w-10 xl:w-12"
                src="help.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2  grow lg:rounded-lg"
    >
      <div
        class="flex flex-col portrait:order-2 "
      >
        <div
          class="flex flex-row justify-center space-x-4 p-4 "
        >
          <div
            class=""
          >
            <button
              class="!bg-adleryellow translate-x-[1px] translate-y-[1px] !border-b-2 !border-r-2 !border-transparent justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="avatar-editor-category-tab-0"
            >
              <img
                alt=""
                class="h-8 lg:h-12"
                src="hair.svg"
              />
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="avatar-editor-category-tab-1"
            >
              <img
                alt=""
                class="h-8 lg:h-12"
                src="face.svg"
              />
            </button>
          </div>
        </div>
        <div
          class="p-4 max-h-[77vh] portrait:max-h-[30vh] overflow-auto"
        >
          <mockedavatareditorhaircategory
            controller="undefined"
          />
        </div>
      </div>
      <div
        class="p-4"
      >
        <mockedavatareditorpreview
          class="relative w-full h-full"
        />
      </div>
    </div>
  </div>
</div>
`;
